docs/lang.md

```diff
@@ -259,13 +259,176 @@ The macroexpand-1 function is just like macroexpand except it will not
 recursively expand macros when the result of the argument macro form is itself
 a macro form.
 
+The `gensym` builtin is used to generate a new symbol, which is most often used
+with macros to avoid avoid naming collisions.
+
+## Parens () and braces []
+
+Braces can be used as a substitute for parens. As with parens, an open brace `[`
+must be closed using a close brace `]`. Using parens with braces can improve
+readability. Conventionally, braces are used with `let` to define the bindings.
+
+```lisp
+(let ([x 1]
+      [y 2])
+  (+ x y))      ; evaluates to 3
+```
+
 ## Special Operators
 
 A special operator is like a macro, in that it receives unevaluated arguments,
 but the result of a special operator will not be subsequently evaluated.
 Examples of special operators are `if`, `lambda`, and `quasiquote`.  There is
 no facility within the language for defining special operators.
 
+### cond
+`cond` takes an arbitrary number of arguments called clauses. A clause consists
+of a list of exactly two expressions. The first expression in a clause is a
+condition, and the second expression is the result.
+
+For example,
+
+```lisp
+(cond (condition1 result1)
+	(condition2 result2)
+	...
+	(:else resultN))
+```
+
+The value returned by `cond` is computed as follows: if condition1 is `true?`,
+then return result1; else if condition2 is `true?` then return result2; else
+if ...; else return resultN.
+
+If none of the conditions are true (and there is no :else), then `()` is
+returned.
+
+### let vs let\*
+
+`let` and `let*` are used to create bindings for local variables within a 
+new scope.  `let` bindings cannot refer to each other. `let*` bindings can
+refer to each other (left-to-right, top-to-bottom). The result of the
```

Comment:
It's not clear to me what you mean by left-to-right/top-to-bottom. It would probably be better to say that `let*` bindings in happens left-to-right/top-to-bottom and that they can refer to previously bound symbols.